UnitsNet.Tests/GenericMathExtensionsTests.cs

```diff
@@ -10,19 +12,59 @@ namespace UnitsNet.Tests;
 public class GenericMathExtensionsTests
 {
     [Fact]
-    public void CanCalcSum()
+    public void Sum_Empty_ReturnsTheAdditiveIdentity()
     {
-        Length[] values = { Length.FromCentimeters(100), Length.FromCentimeters(200) };
+        Length[] values = [];
 
-        Assert.Equal(Length.FromCentimeters(300), values.Sum());
+        Assert.Equal(Length.Zero, GenericMathExtensions.Sum(values));
     }
 
     [Fact]
-    public void CanCalcAverage_ForQuantitiesWithDoubleValueType()
+    public void Sum_OneQuantity_ReturnsTheSameQuantity()
     {
-        Length[] values = { Length.FromCentimeters(100), Length.FromCentimeters(200) };
+        IEnumerable<Length> values = [Length.FromCentimeters(100)];
 
-        Assert.Equal(Length.FromCentimeters(150), values.Average());
+        Length sumOfQuantities = GenericMathExtensions.Sum(values);
+
+        Assert.Equal(Length.FromCentimeters(100), sumOfQuantities);
+    }
+
+    [Fact]
+    public void Sum_TwoQuantities_ReturnsTheExpectedSum()
+    {
+        IEnumerable<Length> values = [Length.FromCentimeters(100), Length.FromCentimeters(200)];
+
+        Length sumOfQuantities = GenericMathExtensions.Sum(values);
```

Comment:
The reason I'm not using `values.Sum()` is because when we add the `IVectorQuantity` interface, this is going to start testing the other extension method (which is a _better/more specific_ match for the list of `Length`).